src/components/posts/Post.js

```diff
@@ -72,18 +72,19 @@ export default function Post({ post, setPosts, posts,count, setCount }) {
 				></textarea>
 			)) ||
 				post.content}
-			{loggedInUser.id === post.user.id && (
 				<div className='button-container'>
-					<Button className='delete-btn' size='small' variant='contained' color={isDeleteing ? 'error' : 'primary'} onClick={(event) => deletePostHandler(event, post.id)}>{isDeleteing ? 'Confirm': 'Delete'}</Button>
+				{(loggedInUser.role === 'TEACHER' || loggedInUser.id === post.user.id) &&(
```

Comment:
Can we extract this to a helper function now

Depending on where this function is written, you might need to pass in the logged in user as an argument
Something like:
```js
const acceptedRoles = ['TEACHER']
const isPostDeletableByUser = (post) => {
  return acceptedRoles.includes(loggedInUser.role) || loggedInUser.id === post.user.id
}
```

Reply:
Thank you for reviewing, will implement these changes today.